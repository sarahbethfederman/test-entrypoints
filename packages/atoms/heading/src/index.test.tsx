import * as React from 'react';
import { shallow } from 'enzyme';
import { HeadingColor, HeadingAlignment, H1, H2, H3, H4, H5, H6 } from './index';

const colors = {
  colorBrandPrimary: '#00C0A5',
  colorShadeLight1: '#FFFFFF',
  colorShadeDark2: '#4A4A4A',
};

describe('Heading', () => {
  const allHeadings = [H1, H2, H3, H4, H5, H6];
  const colorProps: HeadingColor[] = ['colorBrandPrimary', 'colorShadeLight1', 'colorShadeDark2'];
  const textAlignProps: HeadingAlignment[] = ['left', 'center', 'right'];
  allHeadings.forEach((Heading, index) => {
    describe(`H${index + 1}`, () => {
      colorProps.forEach((color) => {
        it.only(`should display ${color} color according props`, () => {
          const wrapper = shallow(<Heading color={color} />);
          expect(wrapper).toHaveStyleRule('color', colors[color]);
        });
      });

      textAlignProps.forEach((alignment) => {
        it(`should align text to ${alignment} according props`, () => {
          const wrapper = shallow(<Heading textAlign={alignment} />);
          expect(wrapper).toHaveStyleRule('text-align', alignment);
        });
      });

      it(`should render default color if no color props is passed in`, () => {
        const wrapper = shallow(<Heading />);
        expect(wrapper).toHaveStyleRule('color', colors.colorShadeDark2);
      });

      it(`should render initial alignment if no textAlign props is passed in`, () => {
        const wrapper = shallow(<Heading />);
        expect(wrapper).toHaveStyleRule('text-align', 'initial');
      });

      it(`should have the right H${index + 1} css`, () => {
        const wrapper = shallow(<Heading />);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
