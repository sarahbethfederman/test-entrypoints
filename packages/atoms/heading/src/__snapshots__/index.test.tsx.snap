// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading H1 should have the right H1 css 1`] = `
.c0 {
  font-size: 46px;
  line-height: 56px;
  font-weight: bold;
  color: #4A4A4A;
  text-align: initial;
}

<h1
  className="c0"
/>
`;

exports[`Heading H2 should have the right H2 css 1`] = `
.c0 {
  font-size: 37px;
  line-height: 48px;
  font-weight: semi-bold;
  color: #4A4A4A;
  text-align: initial;
}

<div
  className="c0"
/>
`;

exports[`Heading H3 should have the right H3 css 1`] = `
.c0 {
  font-size: 28px;
  line-height: 36px;
  font-weight: semi-bold;
}

<div
  className="c0"
/>
`;

exports[`Heading H4 should have the right H4 css 1`] = `
.c0 {
  font-size: 22px;
  line-height: 28px;
  font-weight: semi-bold;
  color: #4A4A4A;
  text-align: initial;
}

<div
  className="c0"
/>
`;

exports[`Heading H5 should have the right H5 css 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  font-weight: semi-bold;
  color: #4A4A4A;
  text-align: initial;
}

<div
  className="c0"
/>
`;

exports[`Heading H6 should have the right H6 css 1`] = `
.c0 {
  font-size: 15px;
  line-height: 20px;
  font-weight: semi-bold;
  color: #4A4A4A;
  text-align: initial;
}

<div
  className="c0"
/>
`;
